pallets/settlement/src/lib.rs

```diff
@@ -228,6 +228,11 @@ impl InstructionId {
     }
 }
 
+/// A wrapper for InstructionMemo
+#[derive(Encode, Decode, TypeInfo, VecU8StrongTyped)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+pub struct InstructionMemo(Vec<u8>);
```

Comment:
Based on what we discussed in the stand-up.  Change the memo from a `Vec` to fixed length array:
```suggestion
pub struct InstructionMemo(pub [u8; 32]);
```